tests/test_script.py

```diff
@@ -279,41 +280,90 @@ def test_same_results(self):
+                     'assert 123.456 == pytest.approx(123.5, abs=0.2), "text"')
+        check_passes(refac,
+                     'assert_almost_equals(123.456, 123.5, msg="text", delta=0.3)',
+                     'assert 123.456 == pytest.approx(123.5, abs=0.3), "text"')
+        check_fails(refac,
+                    'assert_almost_equals(123.456, 124, delta=0.1)',
+                    'assert 123.456 == pytest.approx(124, abs=0.1)')
+
+    def test_not_almost_equal(self):
+        check_passes(refac,
+                     'assert_not_almost_equal(123.456, 123.5, delta=0.01)',
+                     'assert 123.456 != pytest.approx(123.5, abs=0.01)')
+        check_passes(refac,
+                     'assert_not_almost_equal(123.456, 123.5, delta=0.02, msg="text")',
+                     'assert 123.456 != pytest.approx(123.5, abs=0.02), "text"')
+        check_passes(refac,
+                     'assert_not_almost_equal(123.456, 123.5, msg="text", delta=0.03)',
+                     'assert 123.456 != pytest.approx(123.5, abs=0.03), "text"')
+        check_fails(refac,
+                    'assert_not_almost_equal(123.456, 124, delta=0.6)',
+                    'assert 123.456 != pytest.approx(124, abs=0.6)')
+
+        check_passes(refac,
+                     'assert_not_almost_equals(123.456, 123.5, delta=0.01)',
+                     'assert 123.456 != pytest.approx(123.5, abs=0.01)')
+        check_passes(refac,
+                     'assert_not_almost_equals(123.456, 123.5, delta=0.02, msg="text")',
+                     'assert 123.456 != pytest.approx(123.5, abs=0.02), "text"')
+        check_passes(refac,
+                     'assert_not_almost_equals(123.456, 123.5, msg="text", delta=0.03)',
+                     'assert 123.456 != pytest.approx(123.5, abs=0.03), "text"')
+        check_fails(refac,
+                    'assert_not_almost_equals(123.456, 124, delta=0.6)',
+                    'assert 123.456 != pytest.approx(124, abs=0.6)')
+
+    def test_places(self):
+        check_transformation('assert_almost_equal(123.456, 124, places=1)',
+                             'assert 123.456 == pytest.approx(124, abs=1e-1)')
+        check_transformation('assert_almost_equal(123.456, 124, places=7)',
+                             'assert 123.456 == pytest.approx(124, abs=1e-7)')
+        check_passes(refac,
+                     'assert_almost_equal(123.456, 123.450, places=1)',
+                     'assert 123.456 == pytest.approx(123.450, abs=1e-1)')
+
+    def test_decimal(self):
+        check_transformation('assert_almost_equal(123.456, 124, decimal=1)',
+                             'assert 123.456 == pytest.approx(124, abs=1e-1)')
+        check_transformation('assert_almost_equal(123.456, 124, decimal=7)',
+                             'assert 123.456 == pytest.approx(124, abs=1e-7)')
+        # check_passes(refac,
+        #              'assert_almost_equal(123.456, 123.450, decimal=1)',
```

Comment:
To use this one, we need to add `numpy`'s `assert_almost_equal`. I need to check how to do that. Meanwhile the transformation is already ok. 

What I'm not sure is the same as in the case of places, which if we compare `123.456` to `123.400` up to the first decimial it fails. Not sure why, maybe I'm not understanding the behavior of decimal or place properly

Reply:
One issue that I had not thought of is that the script is meant to convert nose assertions to pytest assertions. Including assertions from other tools (like numpy) opens a can of worms, and we have to think this through. What is the relationship there, does e.g. numpy explicitely support nose assertions? (in this case the policy could be that any lib that supports nose assertions can be supported).

Reply:
the signature of almost equal is almost the same. [see here](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.testing.assert_almost_equal.html). It only changes `places` parameter by `decimal`, and adds an extra parameter verbose.

Another thing that I was thinking was that it might be configured which assertions to translate. In case you only want to target some specific transformation. For instance to split the changes across commits or simply because you prefer to keep some asserts that had been rewritten by some other package.

Reply:
maybe we can merge to master removing this part of `numpy`, publish the release and keep working on it. We would support `nosetest` and I can add this part on my fork.

Reply:
But what is the motivation for transforming a numpy assertion to a raw assertion? The point of doing that when using pytest is that pytest traps raw assertions and prints useful info, so after the conversion one can eliminate nose as a dependency. 

But numpy assertions are designed to work with numpy data structures (arrays etc) so typically there would not be a good way of expressing them as equivalent raw assertions. For example the numpy.test.assert_almost_equal would almost certainly work with numpy arrays of any dimension, as well as scalars. So it would likely be incorrect to transform all occurrences of numpy's assert_almost_equal. 

I think it makes sense to transform the nose.tools.assert_almost_equal to use pytest's approx, but the numpy one, I just don't see how it belongs in nose2pytest. Are you actually able to use your latest version on your code base to your satisfaction?

Reply:
Actually you are right. Changing numpy's might not bring usefull information. So how do we ensure that it acutally translate only those comming from nosetest?

Reply:
It is not possible at the AST stage to know which module the assert_almost_equal belongs to. There fore, there is no way of preventing inadvertent transformations when the same symbol name is defined in more than one module (such as nose.tools and numpy.tests). There is also no reliable  way of knowing whether the arg is a scalar or numpy array. Based on https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.testing.assert_almost_equal.html, it would therefore be imprudent to transform an assert_almost_equal that has a "places" or "decimals" argument. 

In fact, this whole discussion makes me realize that the code in script.py should be extended to return the assertion node *unchanged* if there are arg names that are not recognized as being those of a nose assertion. This would apply to all nose assertions, not just assert_almost_equal. I'm guessing that such check would be done in the transform(), based on an "allowed arg names" field defined for each assertion, by checking the keys in "results" dict, although part of the check implementation may have to be in the derived class. 

So unless I'm misunderstanding your argument, the only part of your latest commit that can be kept for this PR is the version command line ARG. If you'd like to try the arg names check, please do it as a separate PR.